GitUI/CommandsDialogs/BrowseDialog/FormOpenDirectory.cs

```diff
@@ -98,13 +98,14 @@ private void LoadClick(object sender, EventArgs e)
             if (Directory.Exists(_NO_TRANSLATE_Directory.Text))
             {
                 _chosenModule = new GitModule(_NO_TRANSLATE_Directory.Text);
-                ThreadHelper.JoinableTaskFactory.Run(() => RepositoryHistoryManager.Locals.AddAsMostRecentAsync(_chosenModule.WorkingDir));
-                Close();
-            }
-            else
-            {
-                MessageBox.Show(this, _warningOpenFailed.Text, _warningOpenFailedCaption.Text);
+                if (_chosenModule.IsValidGitWorkingDir())
+                {
+                    ThreadHelper.JoinableTaskFactory.Run(() => RepositoryHistoryManager.Locals.AddAsMostRecentAsync(_chosenModule.WorkingDir));
+                    Close();
+                }
             }
+
+            MessageBox.Show(this, _warningOpenFailed.Text, _warningOpenFailedCaption.Text, MessageBoxButtons.OK, MessageBoxIcon.Error);
```

Comment:
Won't this message be displayed every time now that it is no more in the else statement?

Reply:
fixed, thank you